memindex.go

```diff
@@ -1,26 +1,24 @@
 package kdb
 
-import (
-	"fmt"
-)
+import "errors"
 
 // Struct representing an element in the index. Here we are maintaining a
 // tree structure. So, it's `Values` field only containes in the leaf nodes only
-// `Map` only conatains in root and intermediate nodes only
+// `Children` only conatains in root and intermediate nodes only
 //
 // Here all the data elements are on the lowest level, which are leafs
 type IndexElement struct {
-	Values        []string
-	BlockPosition int64
-	Map           map[string]*IndexElement
+	Values   []string
+	Position int64
+	Children map[string]*IndexElement
 }
 
 // Base struct of the MemIndex
 // It contains an element called `root` which is the starting point of the tree
-// orderedKeys contains all the keys in this index in ordered fashion
+// keys contains all the keys in this index in ordered fashion
 type MemIndex struct {
-	root        *IndexElement
-	orderedKeys []string
+	Root *IndexElement
```

Comment:
Ah Owen. Yes yes.
Btw: that test failures could be fixed by sorting positions array.
On 2015 මැයි 23, සෙන at ප.ව. 5.12 Muhammed Thanish <notifications@github.com>
wrote:

> In memindex.go
> <https://github.com/meteorhacks/kdb/pull/2#discussion_r30944215>:
>
> >  type MemIndex struct {
> > -	root        *IndexElement
> > -	orderedKeys []string
> > +	Root *IndexElement
>
> json/gob encoders ignore private fields
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meteorhacks/kdb/pull/2/files#r30944215>.
>